pkg/console/operator/sync_v400.go

```diff
@@ -123,12 +124,23 @@ func sync_v400(co *consoleOperator, originalOperatorConfig *operatorv1.Console,
 		}
 	}
 
-	// the operand is available if all resources are present & if we have all the replicas
+	// the operand is available if all resources are:
+	// - present
+	// - if we have at least one ready replica
+	// - route is admitted
 	// available is currently defined as "met the users intent"
-	if deploymentsub.IsReady(actualDeployment) && routesub.IsAdmitted(rt) {
-		co.ConditionDeploymentAvailable(operatorConfig)
-	} else {
+	if !deploymentsub.IsReady(actualDeployment) {
 		co.ConditionDeploymentNotAvailable(operatorConfig)
+	} else if !routesub.IsAdmitted(rt) {
+		co.SetStatusCondition(
```

Comment:
What will get tricky, is we could step on ourselves if we don't aggregate these somehow (but we do want to keep the fail early pattern of the sync loop).   We may have several places that set `OperatorStatusTypeAvailable`, but what if they set `True`, `False`, `True`?  But for different reasons?  Any `False` should result in `False` within a single `Sync` loop run, honestly.  (For `Available`, but not necessarily for the others, `Progressing` would be opposite, if one item is `True`, it should be `True`).

That's where I find the named functions easier to sort out because currently we have to keep these manually maintained.  😦